controllers/storagecluster/provider_server.go

```diff
@@ -508,3 +516,31 @@ func (o *ocsProviderServer) createJob(r *StorageClusterReconciler, instance *ocs
 	r.Log.Info("Job is running as desired")
 	return reconcile.Result{}, nil
 }
+
+func (o *ocsProviderServer) updateClientConfigMap(r *StorageClusterReconciler, namespace string) error {
+	clientConfig := &corev1.ConfigMap{}
+	clientConfig.Name = ocsClientConfigMapName
+	clientConfig.Namespace = namespace
+
+	if err := r.Client.Get(r.ctx, client.ObjectKeyFromObject(clientConfig), clientConfig); err != nil {
+		r.Log.Error(err, "failed to get ocs client configmap")
+		return err
+	}
+
+	if clientConfig.Data == nil {
+		clientConfig.Data = map[string]string{}
+	}
+
+	data := clientConfig.Data
+	if data[deployCSIKey] == "true" {
+		return nil
+	}
+
+	data[deployCSIKey] = "true"
+	if err := r.Client.Update(r.ctx, clientConfig); err != nil {
+		r.Log.Error(err, "failed to update ocs client configmap for enabling CSI")
+		return err
+	}
```

Comment:
sry, I don't see the upside of doing a deep equal.

Reply:
Manual tracking of changes is a known surface for bugs. It is easy when you have a single field to edit but it becomes more complicated as additional and maybe optional/conditional changes are needed.

With my suggestion, we have a way to avoid the entire manual change tracking process. So future contributors will just ignore the need to track changes. That would simplify future changes as well as will most likely avoid potential bugs. 

I hope the upside is clear now

Reply:
done.